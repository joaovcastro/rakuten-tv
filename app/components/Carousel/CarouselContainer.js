import styled from 'styled-components';
import PropTypes from 'prop-types';

const CarouselContainer = styled.div`
  display: flex;
  margin: 0;
  transition: ${props => (props.sliding ? 'none' : 'transform 1s ease')};
  transform: ${props => {
    if (!props.sliding) return 'translateX(calc(-15% - 20px))';
    if (props.direction === 'prev')
      return 'translateX(calc(2 * (-15% - 20px)))';
    return 'translateX(0%)';
  }};
`;

CarouselContainer.propTypes = {
  sliding: PropTypes.bool,
  direction: PropTypes.string.isRequired,
};

CarouselContainer.defaultProps = {
  sliding: false,
  direction: '',
};

export default CarouselContainer;
