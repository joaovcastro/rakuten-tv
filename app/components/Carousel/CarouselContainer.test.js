import React from 'react';
import { shallow } from 'enzyme';

import CarouselContainer from './CarouselContainer';

describe('<CarouselContainer />', () => {
  it('correctly renders', () => {
    const wrapper = shallow(<CarouselContainer />);
    expect(wrapper).toMatchSnapshot();
  });
});
