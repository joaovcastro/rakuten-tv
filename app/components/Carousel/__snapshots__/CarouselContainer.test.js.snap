// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CarouselContainer /> correctly renders 1`] = `
<StyledComponent
  direction=""
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": false,
        "rules": Array [
          "
  display: flex;
  margin: 0;
  transition: ",
          [Function],
          ";
  transform: ",
          [Function],
          ";
",
        ],
      },
      "defaultProps": Object {
        "direction": "",
        "sliding": false,
      },
      "displayName": "styled.div",
      "foldedComponentIds": Array [],
      "propTypes": Object {
        "direction": [Function],
        "sliding": [Function],
      },
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
  sliding={false}
/>
`;
